src/Common/tests/System/Net/VirtualNetwork/VirtualNetworkStream.cs

```diff
@@ -92,16 +105,32 @@ public override void Write(byte[] buffer, int offset, int count)
             _network.WriteFrame(_isServer, innerBuffer);
         }
 
-        private void UpdateReadStream()
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
         {
-            if (_readStream != null && (_readStream.Position < _readStream.Length))
+            try
+            {
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
Nit: I think you said this will never be canceled at this time, so this can wait until later, but if this does have cancellation requested, it'll end up throwing a cancellation exception that's then caught by the catch block, which will return a task in the Faulted state.  Technically it'd be better for the task to be in the Canceled state.  In reality, for your testing purposes, it's unlikely to matter, as if the operation is immediately awaited, the caller won't be able to tell the difference.  If, however, a caller were to inspect the Status of the returned task once it completed (as they might if they were using ContinueWith), they could tell the difference.